### PR TITLE
Add missing deps to files-changed, downgrade `markdownlint-cli`

### PR DESCRIPTION
**Pull request by silverwind**
The `docs` and `yaml` actions categories need to run when the dependencies `markdownlin-cli` or `yamllint` change, so add those to the list of dependencies for these actions.

### CODE REVIEW

**Comment by silverwind** [comment]
Also includes a downgrade to `markdownlint-cli` to fix https://github.com/go-gitea/gitea/issues/27098.

**Review by puni9869 — APPROVED** [review]

**Comment by puni9869** [comment]
No blocker once linter is fix. Execute

**Review by CaiCandong — APPROVED** [review]

**Comment by silverwind** [comment]
Merging this now to unblock CI.

**Comment by puni9869** [comment]
Thanks @silverwind for the quick support on this fix

**Comment by silverwind** [comment]
Likely causes in markdownlint:

https://github.com/DavidAnson/markdownlint/commit/a73658895821444792320ae58b3676d3ca042310
https://github.com/DavidAnson/markdownlint/commit/1ddee6b195d7d796de1b985865f5ce06090c25c1

File: .github/workflows/files-changed.yml [hunk 1]
```diff
@@ -64,9 +64,12 @@ jobs:
               - "**/*.md"
               - "docs/**"
               - ".markdownlint.yaml"
+              - "package.json"
+              - "package-lock.json"
```
Comment (delvh) [inline .github/workflows/files-changed.yml:68]: I think those two are a bit weird for the `docs` category.
If I see that correctly, it'll mean that any JS dependency update or new JS feature will be marked as `kind/docs`.

Comment (puni9869) [reply .github/workflows/files-changed.yml:68]: Feel free to add or remove the `kind/docs`

Comment (silverwind) [reply .github/workflows/files-changed.yml:68]: This is not the labels config, this is the actions config. `mardownlint-cli` version is defined in these files, if those change we need to run the docs workflow to confirm the lnting still works.

Comment (puni9869) [reply .github/workflows/files-changed.yml:68]: got it.

**Pull request merged**